### PR TITLE
Feat/optimism goerli deployment

### PR DESCRIPTION
**Pull request by andremury**
### Changelog

- [X] Added `contractAddresses` util to store eas and community addresses for each network in `hardhat.config.ts`
- [X] Changed addresses variable on deploy scripts
- [X] Added deployment artifacts for Optimism Görli

### CODE REVIEW

File: contracts/ReferrerResolver.sol [hunk 1]
```diff
@@ -30,9 +30,9 @@ contract ReferrerResolver is SchemaResolver, Initializable, OwnableUpgradeable {
             require(ref.uid != bytes32(0), "Referred attestation not valid.");
             require(
                 ref.attester == attestation.attester ||
-                    ref.recipient == attestation.attester ||
```
Comment (maheshmurthy) [inline contracts/ReferrerResolver.sol:33]: Curious why this condition was removed?

File: contracts/Gap.sol [hunk 2]
```diff
@@ -248,4 +180,100 @@ contract Gap is Initializable, OwnableUpgradeable, EIP712Upgradeable {
...
+    function _attest(
+        AttestationRequest calldata request,
+        address attester
+    ) private returns (bytes32) {
+        AttestationRequestData[]
+            memory requestData = new AttestationRequestData[](1);
+        requestData[0] = request.data;
+        validateCanAttestToRefs(requestData, attester);
+
+        return eas.attest(request);
```
Comment (maheshmurthy) [inline contracts/Gap.sol:196]: What is stopping someone from bypassing our method and directly calling this?

File: contracts/Gap.sol [hunk 3]
```diff
@@ -248,4 +180,100 @@ contract Gap is Initializable, OwnableUpgradeable, EIP712Upgradeable {
...
+                    request.data[j].uid
+                );
+
+                require(
+                    target.attester == revoker || target.recipient == revoker,
+                    "GAP:Not owner."
+                );
+            }
+        }
+        eas.multiRevoke(multiRequests);
```
Comment (maheshmurthy) [inline contracts/Gap.sol:221]: Same as above, nothing stops someone from calling this directly isn't it?

**Review by maheshmurthy — CHANGES_REQUESTED** [review]

File: contracts/Gap.sol (hunk 2 above)
Comment (andremury) [reply contracts/Gap.sol:196]: The resolver we set on the schema

**Review by maheshmurthy — APPROVED** [review]

**Pull request merged**